### ISSUE TITLE
Example for how to use

### ISSUE THREAD

**Issue by turkeyphant** [ahvlima/rcloneMon#1]
Can't get this working unfortunately.

Using:

```{
    "servers": [ {
                    "name":"user defined server name",                
                    "addr":"http://server.exemple.com:5572",
                    "description":"User defined server description",
                    "speedcap":2883584,
                    "bwidth":{"500K":"500KBps","1M":"1MBps","1.5M":"1.5MBps","2M":"2MBps","2.5M":"2.5MBps","3M":"3MBps","off":"Unlimited"},
                    "logfile":"rclone.log"
    } ],
    "jobs": [ {
                  "name":"job1",
                  "server":"name of the server where job runs",
                  "commnand":"this is the command to be executed to start the job - not implemented yet",
                  "description":"user define description for the job"
    } ]
}
```

on my server server.exemple.com I run `rclone --rc --config /config/rclone.conf copy /data/file.txt  remote:dir -vv -P`

However I can see nothing at either `http://server.exemple.com:5572` or `http://example2.com/rcloneMon.html?job=job1`

What am I missing?

**Comment by ahvlima** [comment]
Let me try to help you:

Using the config.json as supplied will not work. This is not a working config. It’s a template.

- servers “name” must match the job “server”.
- addr on server is the actual address / ip of the host where your instance of rclone is running. example.com <http://example.com/> is a fictitious name used in documentation. 
- You need to set speedcap and bwidth to match your internet link upstream bandwidth


For example: if your host runs at 192.168.1.50 (this is also an example) try:

    "servers": [ {
                    "name”:"myserver",                
                    "addr":"http://192.168.1.50:5572 <http://192.168.1.50:5572/>",
                    "description”:”My RClone instance",
                    "speedcap":2883584,
                    "bwidth":{"500K":"500KBps","1M":"1MBps","1.5M":"1.5MBps","2M":"2MBps","2.5M":"2.5MBps","3M":"3MBps","off":"Unlimited"},
                    "logfile":"rclone.log"
    } ],
    "jobs": [ {
                  "name":"job1",
                  "server”:"myserver",
                  "commnand":"",
                  "description”:”This is my copy job"
    } ]
}


Run rclone with --rc and --rc-addr=0.0.0.0:5572 

This will enable remote control and allow control from anywhere in your network (default is from the local host only).
I hope it helps.

Arnaldo.

**Comment by turkeyphant** [comment]
Thanks for the quick response.

Yes, I changed all `example.com` - only used it here as illustrative examples as intended.

Under `add` I used hostname as `http://example.com` (without the <> part after).

For server name I'm not sure what I'm supposed to put here as it seems it's a custom string. It seemed from the config you can call it what you want. The docs say "user defined server name, used to address the link to the server from the job array" but there's no concept I'm aware of of a server name with rclone?

As explained, I ran `rclone --rc --config /config/rclone.conf copy /data/file.txt remote:dir -vv -P` and cannot see anything even from localhost. What more do I need to add? I have also tried `rclone --rc --rc-addr=0.0.0.0:5572 --config /config/rclone.conf copy /data/file.txt remote:dir -vv -P` with the same result.

And what's the easiest way to debug this? The rc is definitely running as I get "Not found" when going to example.com:5572

**Comment by ahvlima** [comment]
You need --rc-files <path to the directory where all the file are> 

Also, the “server name” is a string to match withj the “job server”.

**Comment by turkeyphant** [comment]
I thought --rc-files was for `rclone rc` not `rclone --tc`.

Could you just give a working `rclone --rc` command to start a transfer and a working config.json and `rcloneMon.html` URI to test with?

I've not edited the config.json so `"name":"user defined server name"` matches with `"server":"name of the server where job runs"`. It's confusing why they didn't in the example.

**Comment by ahvlima** [comment]
Give me your up address and I will send you both the rclone and config.json.

**Comment by turkeyphant** [comment]
192.168.0.123

**Comment by ahvlima** [comment]
Assuming you are placing all files under the “/data” directory and that the config is at the “/config” directory

rclone -v -P --rc --rc-addr=0.0.0.0:5572 --rc-files /data --config /config/rclone.conf copy /data/file.txt remote:dir

 Please note that if “file.txt” is too small, rclone will start and exit before you can open monitor. It’s for long running jobs.

Point the browser to:  http://192.168.0.123:5572/rcloneMon.html?job=job1 <http://192.168.0.123:5572/rcloneMon.html?job=job1>

Using this config.json file.

**Comment by turkeyphant** [comment]
Afraid that still doesn't work. Those are exactly the same as the commands in my post [here](https://github.com/ahvlima/rcloneMon/issues/1#issuecomment-485590504).

But I tried again moving the flags to the order you suggested.

When I go to  http://192.168.0.123:5572 I get "404 page not found"

When I go to http://192.168.0.123:5572/rcloneMon.html?job=job1 I still get

```
rclone job running on http://192.168.0.123:5572
Error loading/processing configuration file (server may be offline)
```

I can see that rclone is launching the remote properly and I can see that from localhost as well as remote hosts. But rcloneMon is not able to speak to it.

Note I am getting `404 page not found` instead of `not found` now.

**Comment by turkeyphant** [comment]
Okay I've done further investigation.

I am using the `config.json` as per your [comment](https://github.com/ahvlima/rcloneMon/issues/1#issuecomment-485583586) with server "myserver" and job "job1".

I am first starting off the job with the command `rclone -v -P --rc --rc-addr=0.0.0.0:5572 --rc-files /data --config /config/rclone.conf copy /data/file.txt remote:dir`

This works by starting the HTTP server at locahost:5572 which is accessible locally and remotely.

I can test this by running `rclone rc core/stats` which produces:

```
# rclone rc core/stats
{
        "bytes": 1269760,
        "checks": 0,
        "deletes": 0,
        "elapsedTime": 13.779999058,
        "errors": 0,
        "fatalError": false,
        "retryError": false,
        "speed": 92145.14418002365,
        "transferring": [
                {
                        "bytes": 1269760,
                        "eta": 758,
                        "name": "file.txt",
                        "percentage": 1,
                        "size": 88768956,
                        "speed": 117141.67522485304,
                        "speedAvg": 115431.56584418438
                }
        ],
        "transfers": 0
```

So the rc server is up and able to report on the running job correctly. However, note the job name is automatically filled as "file.txt" not "job1".

Strangely, if I try `rclone rc job/list` to see all currently running jobs I get

```
{
        "jobids": []
}
```

Which indicates there are no jobs despite the clear job running under  `"transferring": [` above

Also, if I go to `http://192.168.0.123:5572` I'm now getting `Not found` again.

The job is running under "file.txt" not "job1" so I then changed the config.json to list the job as "file.txt" instead. Now if I go to `http://192.168.0.123:5572/rcloneMon.html?job=file.txt` I get `rclone job running on http://192.168.0.123:5572
Error loading/processing configuration file (server may be offline)`

If I try another other job name such as "doesnotexist" I get, as expected `Cannot find job "doesnotexist" in configuration file`.

So the config..json is working fine, the rc server is working fine. It's just not creating a job id and rcloneMon.html is not able to read anything.

In fact, even if I control the remote control with `rclone rc` commands, I don't create any jobs. Just transfers. So if I run `rclone rc sync/cop srcFs=remote:dir1 dstFs=remote:dir2` and then check `core/stats` I get:

```
~] # rclone rc core/stats
{
        "bytes": 62639697,
        "checks": 10,
        "deletes": 0,
        "elapsedTime": 403.221344739,
        "errors": 0,
        "fatalError": false,
        "retryError": false,
        "speed": 155348.1674948182,
        "transferring": [
                {
                        "bytes": 1777664,
                        "eta": 104,
                        "name": "1.txt",
                        "percentage": 11,
                        "size": 15036823,
                        "speed": 142349.22381674687,
                        "speedAvg": 126976.08544537328
                },
                {
                        "bytes": 1478325,
                        "eta": 0,
                        "name": "2.txt",
                        "percentage": 100,
                        "size": 1478325,
                        "speed": 114728.38451673301,
                        "speedAvg": 105594.53420036717
                },
 etc.
```

There's still nothing in ` "jobids": []`

**Comment by ahvlima** [comment]
There are 2 types of jobs:

- ONE SINGLE SYNCHRONOUS job, started in the rclone command line
- Many asynchronous jobs, started thru the remote control

Core/stats list the statistics of ALL running jobs combined.

Job/stats list statistics of ASYNC jobs ONLY. That’s why your are not seeing the job you started on the command like.

The monitor uses the Single synchronous job. 

Back to the config.json: the names are just strings to allow the Javascript to find the proper configuration information:

The name of the server is to allow one to link one or more jobs to a server.

Job(server)=server(name) 

This creates a complete description of the job and where to connect to get the data.

The job name in the config.json is to allow the user to specify the job to display in the URL, line http://191.168.0.123:5572/rcloneMon.html?job=job1 <http://191.168.0.123:5572/rcloneMon.html?job=job1>

The value passed as a parameter in the URL must match the value of the job(name) in the son file.

If you are getting “Error loading/processing configuration file (server may be offline)”, it’s because either there is an error in the json file or rclone is not running (or has already finished).

Hope this helps.

Arnaldo.

**Comment by turkeyphant** [comment]
I'm still just looking for the most basic version of an example command that works.

**Comment by ahvlima** [comment]
rcloneMon uses only core/stats. Job/stats don’t have progress information.

**Comment by turkeyphant** [comment]
I'm using `rclone -v -P --rc --rc-addr=0.0.0.0:5572 --rc-files /data --config /config/rclone.conf copy /data/file.txt remote:dir` to run a single synchronous job as per your instructions.

I can see it under `"transferring": [` using `core/stats` as you state.

And yet under `http://191.168.0.123:5572/rcloneMon.html?job=job1` it doesn't work.

I can see the `rclone copy` still running in the rclone --rc --rc-addr=0.0.0.0:5572 --config /config/rclone.conf copy /data/file.txt remote:dir -vv -P` terminal and I'm using your `config.json` exactly as provided.

Note that the job name produced by your command is still the filename not `job1`.

**Comment by ahvlima** [comment]
What happens when you try to connect with:

 http://191.168.0.123:5572/rcloneMon.html?job=job1 <http://191.168.0.123:5572/rcloneMon.html?job=job1>

As for the job name, my program does not generates a job name: it gets the parameter from the URL and uses it to find the information inside the config.json.

Arnaldo

**Comment by turkeyphant** [comment]
>What happens when you try to connect with:
>
>http://191.168.0.123:5572/rcloneMon.html?job=job1

As mentioned [here ](https://github.com/ahvlima/rcloneMon/issues/1#issuecomment-485803101)and [here](https://github.com/ahvlima/rcloneMon/issues/1#issuecomment-485888872) I get:

```
rclone job running on  http://191.168.0.123:5572/
Error loading/processing configuration file (server may be offline)
```

It now seems to me that this is an error in the `config.json` but I am using this exactly as supplied by you.

Also, since if I try a URL with a non existent job name such as "doesnotexist" I get, as expected `Cannot find job "doesnotexist" in configuration file` in the web browser instead which implies the `config.json` is actually okay.

>my program does not generates a job name: it gets the parameter from the URL 

I know. I'm talking about the job name created by the command you supplied which was `rclone -v -P --rc --rc-addr=0.0.0.0:5572 --rc-files /data --config /config/rclone.conf copy /data/file.txt remote:dir`

In this case, even with `job1` in the `config.json` the job name is `file.txt` when you ask `core/stats`.

**Comment by turkeyphant** [comment]
lol, I solved it:

I read in `readme.md`:

>There is a single dependency: transfer speed graphics use C3.js, as well as its dependency D3.js.

So I used `c3.js` and `d3.js`.

Only just now I was looking through the code and javascript console and there's no error checking for inclusion of these files. Turns out `rcloneMon.html` only includes the minified versions and not `c3.js` and `d3.js` as well.

For some reason I don't understand, line 186 of `rcloneMon.js`
```
setError(`Cannot find job "${jobName}" in configuration file`);
```
is activated by an earlier `if ... then` loop if those files are missing, not just if there's an error in the `config.json` file. I would recommend changing your error checking in `rcloneMon.js` to fix this!

Thanks :)
